dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/ClaimedTaskRestRepository.java

```diff
@@ -0,0 +1,163 @@
+import org.dspace.app.rest.model.PoolTaskRest;
+import org.dspace.app.rest.model.hateoas.ClaimedTaskResource;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.event.Event;
+import org.dspace.workflow.WorkflowException;
+import org.dspace.xmlworkflow.WorkflowConfigurationException;
+import org.dspace.xmlworkflow.factory.XmlWorkflowServiceFactory;
+import org.dspace.xmlworkflow.service.WorkflowRequirementsService;
+import org.dspace.xmlworkflow.service.XmlWorkflowService;
+import org.dspace.xmlworkflow.state.Step;
+import org.dspace.xmlworkflow.state.Workflow;
+import org.dspace.xmlworkflow.state.actions.Action;
+import org.dspace.xmlworkflow.state.actions.WorkflowActionConfig;
+import org.dspace.xmlworkflow.storedcomponents.ClaimedTask;
+import org.dspace.xmlworkflow.storedcomponents.XmlWorkflowItem;
+import org.dspace.xmlworkflow.storedcomponents.service.ClaimedTaskService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the repository responsible to manage PooledTask Rest object
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ */
+
+@Component(PoolTaskRest.CATEGORY + "." + ClaimedTaskRest.NAME)
+public class ClaimedTaskRestRepository extends DSpaceRestRepository<ClaimedTaskRest, Integer> {
+
+    private static final Logger log = Logger.getLogger(ClaimedTaskRestRepository.class);
+
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    EPersonService epersonService;
+
+    @Autowired
+    ClaimedTaskService claimedTaskService;
+
+    @Autowired
+    ClaimedTaskConverter converter;
+
+    @Autowired
+    XmlWorkflowService workflowService;
+
+    @Autowired
+    WorkflowRequirementsService workflowRequirementsService;
+
+    @Override
+    public ClaimedTaskRest findOne(Context context, Integer id) {
+        ClaimedTask task = null;
+        try {
+            task = claimedTaskService.find(context, id);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        if (task == null) {
+            return null;
+        }
+        return converter.fromModel(task);
+    }
+
+    @SearchRestMethod(name = "findByUser")
+    public Page<ClaimedTaskRest> findByUser(@Parameter(value = "uuid") UUID userID, Pageable pageable) {
+        List<ClaimedTask> tasks = null;
+        try {
+            Context context = obtainContext();
+            EPerson ep = epersonService.find(context, userID);
+            tasks = claimedTaskService.findByEperson(context, ep);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
We should return a more useful error here. Something like `"Could not find tasks for user with ID" + userID.`

Reply:
see other comments about exception

Reply:
Withdrawing my request & resolving this conversation.